Marketplace/shapes/shapes.js

```diff
@@ -1828,30 +1832,75 @@
     }
 
 
-    function onAppButtonClicked() {
-        var NOTIFICATIONS_MESSAGE_CHANNEL = "Hifi-Notifications",
+    function onTabletButtonClicked() {
+        if (isTabletUIOpen) {
+            tablet.gotoHomeScreen();
+        } else {
+            tablet.gotoWebScreen(APP_UI_HTML + "?active=" + isAppActive);
+        }
+    }
+
+    function onTabletWebEventReceived(data) {
+        var message,
+            SET_ACTIVE_MESSAGE = "setActive",  // EventBridge message.
+            NOTIFICATIONS_MESSAGE_CHANNEL = "Hifi-Notifications",
             EDIT_ERROR = 4, // Per notifications.js.
             INSUFFICIENT_PERMISSIONS_ERROR_MSG =
                 "You do not have the necessary permissions to edit on this domain."; // Same as edit.js.
 
-        // Application tablet/toolbar button clicked.
-        if (!isAppActive && !(Entities.canRez() || Entities.canRezTmp())) {
-            Feedback.play(dominantHand, Feedback.GENERAL_ERROR);
-            Messages.sendLocalMessage(NOTIFICATIONS_MESSAGE_CHANNEL, JSON.stringify({
-                message: INSUFFICIENT_PERMISSIONS_ERROR_MSG,
-                notificationType: EDIT_ERROR
-            }));
+        try {
+            message = JSON.parse(data);
+        } catch (e) {
             return;
         }
 
-        isAppActive = !isAppActive;
-        updateControllerDispatcher();
-        button.editProperties({ isActive: isAppActive });
+        switch (message.command) {
+            case SET_ACTIVE_MESSAGE:
+                switch (message.value) {
+                    case true:
```

Comment:
dual switching? Only two options, `true` and `false`. What made you choose to have a switch statement over simply having an if-else statement?

Reply:
@thoys Defensive programming: `message` is parsed from `data` being a JSON string, and so `message.value` could in theory be `true`, `false`, or `undefined`. In practice, unless there's a code change that produces a bug in the `data` value, it will always be `true` or `false`. Easy to change to an `if()` statement if you still want.